### PR TITLE
[ENDPOINT] UPDATE/activities - OT102 - 67

### PR DESCRIPTION
**Pull request by Gabriiel99**
**Description**
AS admin user
I WANT to update an existing activity
TO keep information up to date

**Criteria of acceptance:**
PUT / activities /: id - You must validate that the activity exists based on the id sent by parameter. In the event that there is no return an error, otherwise update it and return it with the updated data

### CODE REVIEW

File: routes/activities.js [hunk 1]
```diff
@@ -13,5 +13,5 @@ router.get('/', get)
 router.get('/:id', getActivity)
```
Comment (JonatanGarbuyo) [inline routes/activities.js:13]: falta el 
router.get('/', get)

se perdio' en los merges?

File: routes/activities.js [hunk 2]
```diff
@@ -13,5 +13,5 @@ router.get('/', get)
...
 router.delete('/:id', isAdmin, destroy)
 router.post('/', [isAdmin, validateRequest(activitySchema)], post)
-
+router.put('/:id', isAdmin, update)
```
Comment (dvelezroman) [inline routes/activities.js]: falta el middleware para validar los paramtros

**Review by dvelezroman — CHANGES_REQUESTED** [review]

File: services/activities.js [hunk 3]
```diff
@@ -60,3 +61,23 @@ exports.addActivity = async (data) => {
...
+    }
+    activity.set({
+      id, name, image, content,
+    })
+    activity.save()
+
+    return activity
+  } catch (error) {
+    if (error.message === 'ID not found.') {
+      throw new ErrorObject('activity not found with that ID', 404)
```
Comment (gabodmc) [inline services/activities.js:81]: Fijate de cambiarle al nuevo handler `throw Error('activity not found with that ID', 404)`

**Review by gabodmc — CHANGES_REQUESTED** [review]

File: routes/activities.js [hunk 4]
```diff
@@ -13,5 +13,5 @@ router.get('/', get)
 router.get('/:id', getActivity)
 router.delete('/:id', isAdmin, destroy)
 router.post('/', [isAdmin, validateRequest(activitySchema)], post)
-
+router.put('/:id', [isAdmin, validateRequest(activitySchema)], update)
```
Comment (dvelezroman) [inline routes/activities.js:16]: good

**Review by dvelezroman — APPROVED** [review]

**Review by gabodmc — APPROVED** [review]

**Review by gabyMangoni — APPROVED** [review]
Ok

File: services/activities.js [hunk 5]
```diff
@@ -60,3 +60,23 @@ exports.addActivity = async (data) => {
...
+      throw Error(`ID ${id} not found.`)
+    }
+    activity.set({
+      id, name, image, content,
+    })
+    activity.save()
+
+    return activity
+  } catch (error) {
+    if (error.message === 'ID not found.') {
```
Comment (JonatanGarbuyo) [inline services/activities.js:80]: el if no va a coincidir porque el mensaje de error es distinto.  trae el id.

                 throw Error(`ID ${id} not found.`)
if (error.message === 'ID not found.') {

**Review by JonatanGarbuyo — CHANGES_REQUESTED** [review]

File: routes/activities.js (hunk 1 above)
Comment (gabyMangoni) [reply routes/activities.js:13]: esta en la linea 12..

**Review by dvelezroman — APPROVED** [review]

**Review by JonatanGarbuyo — APPROVED** [review]

**Pull request merged**